spec/lang/step/terminators.md

```diff
@@ -85,14 +85,19 @@ fn check_abi_compatibility(
         (Type::Ptr(_), Type::Ptr(_)) =>
             // The kind of pointer and pointee details do not matter for ABI.
             true,
-        (Type::Tuple { fields: caller_fields, size: caller_size, align: caller_align },
-         Type::Tuple { fields: callee_fields, size: callee_size, align: callee_align }) =>
+        (Type::Tuple { sized_fields: caller_fields, sized_head_layout: caller_head_layout, unsized_field: caller_unsized_field },
+         Type::Tuple { sized_fields: callee_fields, sized_head_layout: callee_head_layout, unsized_field: callee_unsized_field }) => {
+            let (caller_size, caller_align) = caller_head_layout.head_size_and_align();
+            let (callee_size, callee_align) = callee_head_layout.head_size_and_align();
+            caller_unsized_field.is_none() &&
+            callee_unsized_field.is_none() &&
```

Comment:
These could be assertions, right? Unsized types cannot currently be passed as arguments.